plugins/gradle/src/index.ts

```diff
@@ -238,8 +242,89 @@ export default class GradleReleasePluginPlugin implements IPlugin {
       ]);
     });
 
+    auto.hooks.canary.tapPromise(
+      this.name,
+      async ({ dryRun, quiet }) => {
+        this.canaryRelease = true;
+
+        const releaseVersion = await getVersion(
+          this.options.gradleCommand,
+          this.options.gradleOptions
+        );
+
+        const canaryVersion = `${releaseVersion}${defaultSnapshotSuffix}`;
+
+        if (dryRun) {
+          if (quiet) {
+            console.log(canaryVersion);
+          } else {
+            auto.logger.log.info(`Would have published: ${releaseVersion}`);
+          }
+
+          return canaryVersion;
+        }
+
+        const { publish } = this.properties;
+
+        if (publish) {
+          await execPromise(this.options.gradleCommand, [
+            "publish",
+            ...this.options.gradleOptions,
+          ]);
+        }
+
+        return canaryVersion;
+      }
+    );
+
+    auto.hooks.next.tapPromise(
+      this.name,
+      async (preReleaseVersions, { dryRun }) => {
+        const branch = getCurrentBranch() || "";
+
+        const version = await getVersion(
+          this.options.gradleCommand,
+          this.options.gradleOptions
+        );
+
+        const nextVersion = `${version}${defaultSnapshotSuffix}`;
+
+        // Don't need to increment version since version should have already 
+        // been incremented by canary during PR, so take the current version
+        // and publish it
```

Comment:
while in some build systems this might be true the `next` hook should not assume that a the version has already been incremented. You'll probably need all of [this logic](https://github.com/intuit/auto/blob/main/plugins/git-tag/src/index.ts#L81-L93) to properly calculate the next version